### ISSUE TITLE
ValueError: vector::_M_default_append

### ISSUE THREAD

**Issue by romanovzky** [scikit-hep/pyhepmc#14]
Hi there,
I was trying out this library in python using with a file I generated. This is the code I ran:
```python
import pyhepmc_ng

adapter = pyhepmc_ng.ReaderAsciiHepMC2(filename="file.hepmc")

evt = pyhepmc_ng.GenEvent(
    momentum_unit=pyhepmc_ng.Units.GEV, length_unit=pyhepmc_ng.Units.MM
)

adapter.read_event(evt)
```
which returns the error
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-12-9f5860f373ad> in <module>
      7 )
      8 
----> 9 adapter.read_event(evt)

ValueError: vector::_M_default_append
```

Disclaimer: I am not versed in hepmc, I was following the logic from [here](http://hepmc.web.cern.ch/hepmc/HepMC2_reader_example_8cc-example.html).

Cheers

**Comment by HDembinski** [comment]
Hi, thanks for trying this out. Unfortunately, we couldn't turn this project into the default Python API for HepMC3, they have their own independent Python bindings. Therefore, I cannot offer much support. I don't see something that you did wrong at first glance, it looks like a bug in our library.

tl;dr: I do not have time to support this library at the moment, so if you are doing serious work, I would recommend the official HepMC3 Python API for now.

**Comment by romanovzky** [comment]
Hi @HDembinski thank you very much for your reply.

I'm sadden to hear that, I was looking forward to having a python interface that wouldn't required installing HepMC...

All the best!

**Comment by HDembinski** [comment]
@romanovzky For my curiosity, did you find a solution or did you switch to the official HepMC3 Python bindings?

**Comment by romanovzky** [comment]
Hi @HDembinski neither to be honest. I was attempting this route to avoid having to use C++, but I just went on with it. I'm still very keen on a working python interface/wrapper, as almost all my work and workflow is strictly pythonic.

---

Also, I was bound to hepmc2 files, and the only hepmc3 seemed to have working python3 bindings?

**Comment by HDembinski** [comment]
Thanks for the heads up. I think you can read the HepMC2 format with HepMC3.

**Comment by romanovzky** [comment]
Now that you say that, I think I tried to use HepMC3 python bindings but had a similar issue. I can't recall the details as I was already not really convinced about the approach anymore. In any case, I did need to put together a FastJet routine, so using C++ was the obvious way. I just wanted to explore a python approach.

**Comment by HDembinski** [comment]
I cannot reproduce this in the latest version, so hopefully updating will fix things for you.
